packages/oidc-client/src/initWorker.ts

```diff
@@ -113,6 +113,11 @@ export const initWorkerAsync = async (
     registration = await configuration.service_worker_register(serviceWorkerRelativeUrl);
   } else {
     registration = await navigator.serviceWorker.register(serviceWorkerRelativeUrl);
+
+    if (registration.active && registration.waiting) {
+      console.log('Detected new service worker waiting, unregistering and reloading');
+      await configuration.service_worker_update_require_callback?.(registration, stopKeepAlive);
```

Comment:
hi  @wermanoid thank you for your PullRequest !
I'am not sure to understand why we nee 2 time  await configuration.service_worker_update_require_callback?.(registration, stopKeepAlive); ? may you help me to understand?

Reply:
there are cases when new service worker is added, when old one is still running. This leads to case when SW is just hanging in `waitting` state and never starts, as original SW is still running as usual.

As far as I understand now, oidc is able to update SW only when running SW version and oidc version mismatch.
But we had cases when ui was redeployed and that leads to case, when 2 SW of same internal oidc version appeared in browser.

Reply:
Thanh you @wermanoid I will test your version tomorrow! sorry for the delay.

Reply:
Hi @guillaume-chervet , any chance to verify? Or can I help somehow to move forward?

Reply:
I checked it referring to my Issue [#1400](https://github.com/AxaFrance/oidc-client/issues/1400)

I can't reproduce the sw mismatched versions issue anymore in demo app running from [wermanoid:fix/update-sw-registration](https://github.com/wermanoid/oidc-client/tree/fix/update-sw-registration)